GUIDELINES.md

```diff
@@ -0,0 +1,994 @@
+    .requireReal<LoginViewModel>()
+```
+
+"require" here means that that you expect the type (in this case `LoginViewModel`) in its real form, not as a mock. So if the configuration says `mockOnly of<LoginViewModel>` or in another steps or test class you declare `requireMock<LoginViewModel>` there's a conflict and sweetest will throw an exception. Expectations about dependencies need to be unanbiguous in a test system.
+
+**Caution:** make sure you have declared the dependency type in the [module testing configuration](#adding-dependencies)!
+
+##### Offering instances
+
+If the standard mock and real instance creation does not work out for you feel free to use the `offerX` class of functions:
+
+```kotlin
+offerReal { AuthManager(myBackendGateway, mySessionStore) }
+offerMock { FakeAuthManager() }
+```
+
+This effectively tells sweetest how a certain type is initialized.
+
+It might make sense to satisfy the constructor of `AuthManager` the way it's shown above, but in most cases you should not circumvent sweetest's dependency management. So you can use `instanceOf()` which is available in the receiver scope of the lambda you pass to the `offerX` functions:
+
+```kotlin
+offerReal { AuthManager(instanceOf(), instanceOf()) }
+```
+
+You might also want to retrieve a more specific type of a certain argument:
+
+```kotlin
+offerReal { AuthManager(instanceOf<CustomBackendGateway>(), instanceOf()) }
+```
+
+That way you tell sweetest's dependency management to retrieve a dependency of type `CustomBackendGateway` rather the one directly specified in `AuthManager`'s constructor.
+
+##### Offering _and_ requiring at the same time
+
+Offering is a different concept than requiring: that you offer the framework how to come to a certain dependency type doesn't mean it's configured to deliver it. E.g. you may `offerReal<AuthManager> { ... }` but forget to call `requireReal<AuthManager>()` somewhere in your test system. As a consequence you'll end up with a mock nevertheless.
+
+But in order to avoid this problem consider using just the `offerXRequired` family of functions, as it caters for the most common use case:
+
+```kotlin
+offerRealRequired<AuthManager> { AuthManager(myBackendGateway, mySessionStore) }
+offerMockRequired<AuthManager> { FakeAuthManager() }
+```
+
+The reason for being able to just "offer" something without actually putting it to use is that steps classes can also be seen as libraries which define how to create a real or mocked version of a type, where the decision what to pick finally can happen elsewhere. To keep things simple, in most cases `offerMockRequired` and `offerRealRequired` does the trick though.
+
+##### Distinguish mock and real correctly
+
+Make sure to distinguish mock and real correctly: when you're using the production type as it's used in your product, consider it real. Everything else that is specific to your test system like mocks, spies and fakes should fall under the "mock" category for sweetest.
```

Comment:
I first read it like _"how do I know if an given instance is a mock or real"_. But what it's actually about: _"When should I use real mode and when mock mode"_. Maybe this makes it a little clearer?:
```suggestion
Make sure to use mock and real modes correctly: when you're using the production type as it's used in your product, consider it real. Everything else that is specific to your test system like mocks, spies and fakes should fall under the "mock" category for sweetest.
```

Reply:
i moved it into the heading